### ISSUE TITLE
Extend MAPL_ConfigSetAttribute to include array of integers as input

### ISSUE THREAD

**Issue by gmao-esherman** [GEOS-ESM/MAPL#660]
As part of refactoring GOCART2G, we need the capability to add a list of integers to a config object using ESMF_ConfigSetAttribute. ESMF_ConfigSetAttribute currently only accepts a single integer as the value argument. Please let me know if you need a description of the use case or any additional information. Thanks.

**Comment by mathomp4** [comment]
I'm assigning to @atrayano or @bena-nasa but if this needs to be in ESMF, we might have to contact...ESMF.

**Comment by atrayano** [comment]
Ultimately ESMF needs to support this, but I have extended similarly ConfigSetAttribute for variety of overloads. I discussed this topic briefly with @tclune yesterday, and I plan to address it as part of another change I am doing for @gmao-esherman (and Arlindo)

**Comment by bena-nasa** [comment]
You can do this, esmf can read a label with multiple values, this just not one step. We would have to write a wrapper to do this. See the documentation:
```
46.2.4 How to retrieve a label with multiple values
When there are multiple, mixed-typed values associated with a label, the values can be retrieved in two steps: 1) Use ESMF_ConfigFindLabel() to find the label in the Config class; 2) use ESMF_ConfigGetAttribute() without the optional 'label' argument to retrieve the values one at a time, reading from left to right in the record.
A second reminder that the order in which a particular label/value pair is retrieved is not dependent upon the order which they exist within the Resource File. The label used in this method allows the user to skip to any point in the file.

      call ESMF_ConfigFindLabel(cf, 'constants:', rc=rc) ! Step a) Find the 
                                                         ! label
Two constants, radius and i_n, can now be retrieved without having to specify their label or use an array. They are also different types.

      call ESMF_ConfigGetAttribute(cf, param_1, rc=rc) ! Step b) read in the 
                                                       ! first constant in 
                                                       ! the sequence
      call ESMF_ConfigGetAttribute(cf, i_n, rc=rc)     ! Step c) read in the 
                                                       ! second constant in 
                                                       ! the sequence
```

**Comment by atrayano** [comment]
Addressed in PR#727, already merged in develop
